readthedocs/builds/storage.py

```diff
@@ -102,6 +103,21 @@ def copy_directory(self, source, destination):
                 with safe_open(filepath, "rb") as fd:
                     self.save(sub_destination, fd)
 
+    def _check_suspicious_path(self, path):
+        """Check that the given path isn't a symlink or outside the doc root."""
+        path = Path(path)
+        resolved_path = path.resolve()
+        if path.is_symlink():
+            msg = "Suspicious operation over a symbolic link."
+            log.error(msg, path=str(path), resolved_path=str(resolved_path))
+            raise SuspiciousFileOperation(msg)
```

Comment:
We should come back to this exception and raise custom ones that communicate the error in a better way. Also, these exception will automatically be catch at `on_failure` from the Celery task and showing a proper error to the user if we use custom ones.